Micronaut OpenAPI server (api-first) generates code with missing imports

### Expected Behavior

I'm following the [Micronaut Server generation with OpenAPI](https://guides.micronaut.io/latest/micronaut-openapi-generator-server-maven-java.html) guide. I checked out the example project, added a property `createdAt: {type: date-time}` to the BookInfo in library-definition.yml. I expect `mvn clean install` to generate the abstract server class.

### Actual Behaviour

The generated code uses `ZonedDateTime`, but the import is missing, causing a compilation error:

```
[ERROR] BookInfo.java:[203,31] cannot find symbol
[ERROR]   symbol:   class ZonedDateTime
[ERROR]   location: class micronaut.example.model.BookInfo
```

### Steps To Reproduce

1. Download the project mentioned at [Micronaut Server generation with OpenAPI](https://guides.micronaut.io/latest/micronaut-openapi-generator-server-maven-java.html).
2. Add a property `createdAt: {type: date-time}` to the BookInfo in library-definition.yml.
3. Run `mvn clean install`.

### Environment Information

_No response_

### Example Application

_No response_

### Version

Latest

--- a/openapi-generator/src/main/java/io/micronaut/openapi/generator/AbstractMicronautJavaCodegen.java
+++ b/openapi-generator/src/main/java/io/micronaut/openapi/generator/AbstractMicronautJavaCodegen.java
@@ -617,15 +617,27 @@ public abstract class AbstractMicronautJavaCodegen<T extends GeneratorOptionsBui
         switch (dateLibrary) {
             case OPT_DATE_LIBRARY_OFFSET_DATETIME -> {
                 typeMapping.put("DateTime", "OffsetDateTime");
+                typeMapping.put("date-time", "OffsetDateTime");
                 typeMapping.put("date", "LocalDate");
+                importMapping.put("DateTime", "java.time.OffsetDateTime");
+                importMapping.put("date-time", "java.time.OffsetDateTime");
+                importMapping.put("date", "java.time.LocalDate");
             }
             case OPT_DATE_LIBRARY_ZONED_DATETIME -> {
                 typeMapping.put("DateTime", "ZonedDateTime");
+                typeMapping.put("date-time", "ZonedDateTime");
                 typeMapping.put("date", "LocalDate");
+                importMapping.put("DateTime", "java.time.ZonedDateTime");
+                importMapping.put("date-time", "java.time.ZonedDateTime");
+                importMapping.put("date", "java.time.LocalDate");
             }
             case OPT_DATE_LIBRARY_LOCAL_DATETIME -> {
                 typeMapping.put("DateTime", "LocalDateTime");
+                typeMapping.put("date-time", "LocalDateTime");
                 typeMapping.put("date", "LocalDate");
+                importMapping.put("DateTime", "java.time.LocalDateTime");
+                importMapping.put("date-time", "java.time.LocalDateTime");
+                importMapping.put("date", "java.time.LocalDate");
             }
             default -> {
             }

--- a/openapi-generator/src/main/java/io/micronaut/openapi/generator/AbstractMicronautKotlinCodegen.java
+++ b/openapi-generator/src/main/java/io/micronaut/openapi/generator/AbstractMicronautKotlinCodegen.java
@@ -396,7 +396,7 @@ public abstract class AbstractMicronautKotlinCodegen<T extends GeneratorOptionsB
         typeMapping.put("binary", "ByteArray");
         typeMapping.put("AnyType", "Any");
         typeMapping.put("DateTime", "Instant");
-        typeMapping.put("date-time", "OffsetDateTime");
+        typeMapping.put("date-time", "Instant");
         typeMapping.put("date", "LocalDate");
         typeMapping.put("Date", "LocalDate");
         typeMapping.put("LocalDateTime", "LocalDateTime");
@@ -654,15 +654,27 @@ public abstract class AbstractMicronautKotlinCodegen<T extends GeneratorOptionsB
         switch (dateLibrary) {
             case OPT_DATE_LIBRARY_OFFSET_DATETIME -> {
                 typeMapping.put("DateTime", "OffsetDateTime");
+                typeMapping.put("date-time", "OffsetDateTime");
                 typeMapping.put("date", "LocalDate");
+                importMapping.put("DateTime", "java.time.OffsetDateTime");
+                importMapping.put("date-time", "java.time.OffsetDateTime");
+                importMapping.put("date", "java.time.LocalDate");
             }
             case OPT_DATE_LIBRARY_ZONED_DATETIME -> {
                 typeMapping.put("DateTime", "ZonedDateTime");
+                typeMapping.put("date-time", "ZonedDateTime");
                 typeMapping.put("date", "LocalDate");
+                importMapping.put("DateTime", "java.time.ZonedDateTime");
+                importMapping.put("date-time", "java.time.ZonedDateTime");
+                importMapping.put("date", "java.time.LocalDate");
             }
             case OPT_DATE_LIBRARY_LOCAL_DATETIME -> {
                 typeMapping.put("DateTime", "LocalDateTime");
+                typeMapping.put("date-time", "LocalDateTime");
                 typeMapping.put("date", "LocalDate");
+                importMapping.put("DateTime", "java.time.LocalDateTime");
+                importMapping.put("date-time", "java.time.LocalDateTime");
+                importMapping.put("date", "java.time.LocalDate");
             }
             default -> {
             }

--- a/openapi-generator/src/test/java/io/micronaut/openapi/generator/JavaMicronautServerCodegenTest.java
+++ b/openapi-generator/src/test/java/io/micronaut/openapi/generator/JavaMicronautServerCodegenTest.java
@@ -768,4 +768,16 @@ class JavaMicronautServerCodegenTest extends AbstractMicronautCodegenTest {
                 private ZonedDateTime date;
             """);
     }
+
+    @Test
+    void testImportZonedDateTime() {
+
+        var codegen = new JavaMicronautServerCodegen();
+        String outputPath = generateFiles(codegen, "src/test/resources/3_0/library-definition.yml", CodegenConstants.APIS, CodegenConstants.MODELS);
+        String path = outputPath + "src/main/java/org/openapitools/";
+
+        assertFileContains(path + "model/BookInfo.java",
+            "import java.time.ZonedDateTime;",
+            "private ZonedDateTime createdAt;");
+    }
 }

--- a/openapi-generator/src/test/java/io/micronaut/openapi/generator/KotlinMicronautServerCodegenTest.java
+++ b/openapi-generator/src/test/java/io/micronaut/openapi/generator/KotlinMicronautServerCodegenTest.java
@@ -903,4 +903,16 @@ class KotlinMicronautServerCodegenTest extends AbstractMicronautCodegenTest {
                 var date: ZonedDateTime? = null,
             """);
     }
+
+    @Test
+    void testImportZonedDateTime() {
+
+        var codegen = new KotlinMicronautServerCodegen();
+        String outputPath = generateFiles(codegen, "src/test/resources/3_0/library-definition.yml", CodegenConstants.APIS, CodegenConstants.MODELS);
+        String path = outputPath + "src/main/kotlin/org/openapitools/";
+
+        assertFileContains(path + "model/BookInfo.kt",
+            "import java.time.ZonedDateTime",
+            "var createdAt: ZonedDateTime? = null,");
+    }
 }

--- a/openapi-generator/src/test/resources/3_0/library-definition.yml
+++ b/openapi-generator/src/test/resources/3_0/library-definition.yml
@@ -1,0 +1,81 @@
+#tag::info[]
+openapi: 3.0.0 # <1>
+info: # <2>
+  description: This is a library API
+  version: 1.0.0
+  title: Library
+  license:
+    name: Apache-2.0
+    url: "https://www.apache.org/licenses/LICENSE-2.0.html"
+tags: # <3>
+  - name: books
+    description: Search for books and add new ones
+#end::info[]
+#tag::paths[]
+paths:
+#end::paths[]
+#tag::search[]
+  /search:
+    get: # <1>
+      tags: 
+        - books # <2>
+      summary: Search for a book
+      operationId: search # <3>
+      parameters: # <4>
+        - name: book-name
+          in: query
+          schema: 
+            type: string
+            minLength: 3 # <5>
+        - name: author-name
+          in: query
+          schema: 
+            type: string
+      responses: # <6>
+        "200": # <7>
+          description: Success
+          content:
+            "application/json":
+              schema:
+                type: array, 
+                items: 
+                  $ref: "#/components/schemas/BookInfo"
+        "400": # <8>
+          description: Bad Request
+#end::search[]
+#tag::add[]
+  /add:
+    post: # <1>
+      tags: [books]
+      summary: Add a new book
+      operationId: addBook
+      requestBody: # <2>
+        required: true
+        content:
+          "application/json": 
+            schema: 
+              $ref: "#/components/schemas/BookInfo" # <3>
+      responses:
+        "200":
+          description: Success
+        "400":
+          description: Bad Request
+#end::add[]
+#tag::components[]
+components:
+  schemas:
+    BookInfo:
+      title: Book Info # <1>
+      description: Object containg all the info about a book
+      type: object
+      properties: # <2>
+        name: {type: string}
+        availability: {$ref: "#/components/schemas/BookAvailability"} # <3>
+        author: {type: string, minLength: 3}
+        isbn: {type: string, pattern: "[0-9]{13}"}
+        createdAt: {type: date-time}
+      required: ["name", "availability"]
+    BookAvailability: # <4>
+      type: string
+      enum: ["available", "not available", "reserved"]
+#end::components[]

